Taskfile.yml

```diff
@@ -29,39 +33,132 @@ tasks:
       - mv {{.BINARY}} {{.LOCATION}}/{{.BINARY}}
 
+  ci:
+    desc: Workflow to run in CI
+    cmds:
+      - task: go:setup
+      - task: go:has-latest-opslevel-go
+      - task: lint
+      - task: test
+
+  clean:
+    desc: Clean up terraform files from "{{.WORKSPACE_DIR}}"
+    dir: "{{.WORKSPACE_DIR}}"
+    prompt: Remove '*.tfstate .terraform.lock.hcl ./terraform.* .terraform/**' from '{{.WORKSPACE_DIR}}' directory?
+    cmds:
+      - cmd: rm -rf *.tfstate .terraform.lock.hcl ./terraform.* .terraform/
+        ignore_error: true
+
+  destroy:
+    desc: Run "terraform destroy" in "{{.WORKSPACE_DIR}}"
+    cmds:
+      - task: terraform-command
+        vars: { TF_COMMAND: "destroy", TF_CMD_DIR: "{{.WORKSPACE_DIR}}" }
+
+  fix:
+    desc: Fix formatting and linting
+    cmds:
+      - task: go:fix
+      - task: format-fix
+
+  init:
+    desc: Initialize terraform workspace
+    dir: "{{.WORKSPACE_DIR}}"
+    cmds:
+      - task: install-terraform
+      - './make_backend_tf.sh'
+      - touch main.tf
+      - task: terraform-command
+        vars: { TF_COMMAND: "init -upgrade", TF_CMD_DIR: "{{.WORKSPACE_DIR}}" }
+
   install-changie:
     desc: Install changie
     aliases: [get-changie]
     cmds:
       - cmd: echo "Installing changie..."
         silent: true
-      - go install github.com/miniscruff/changie@latest
-      - cmd: echo -e "\nSuccess! 'changie' installed."
+      - which changie > /dev/null || go install github.com/miniscruff/changie@latest
+      - cmd: echo -e "Success! 'changie' installed."
         silent: true
 
   lint:
     desc: Check formatting and linting
     cmds:
       - task: go:lint
-      - task: terraform:lint
+      - task: validate
+      - task: format-check
 
-  lintfix:
-    desc: Fix formatting and linting
+  plan:
+    desc: Run "terraform plan" in "{{.WORKSPACE_DIR}}"
     cmds:
-      - task: go:lintfix
-      - task: terraform:lintfix
+      - task: terraform-command
+        vars: { TF_COMMAND: "plan", TF_CMD_DIR: "{{.WORKSPACE_DIR}}" }
 
   setup:
     desc: Setup env and tools
     cmds:
       - task: install-changie
       - task: go:setup
       - task: build
-      - task: terraform:init
+      - task: init
 
   test:
     desc: Run tests
     env:
       TF_ACC: true
     cmds:
       - task: go:test
+
+  ########################################
+  # internal (not directly called) tasks #
+  ########################################
+
+  format-check:
+    internal: true
+    desc: Run terraform format
+    cmds:
+      - cmd: echo "Listing all terraform files that need formatting..."
+      - task: terraform-command
+        vars: { TF_COMMAND: "fmt -recursive -check", TF_CMD_DIR: "{{.ROOT_DIR}}" }
+
+  format-fix:
+    internal: true
+    desc: Run terraform format
+    cmds:
+      - task: terraform-command
+        vars: { TF_COMMAND: "fmt -recursive -write=true", TF_CMD_DIR: "{{.ROOT_DIR}}" }
+
+  install-terraform:
+    internal: true
+    aliases: [get]
```

Comment:
Why does this have the alias `get`?

Reply:
Also I am pro just removing this altogether and replacing it with just `install-terraform-mac-os`. I don't think we're ever going to support other operating systems and if someone wants to contribute on another OS they can handle installation themselves.

Reply:
We might support other OSes down the road. But if we do, we should do so by via containers